### PR TITLE
ECS deployment automation

### PR DESCRIPTION
**Pull request by chicago-joe**
## Description

What's cooking? 
 added deploy script to automate policy / role / service / cluster creation

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

## New Recipe Checklist

- [ ] My PR is in the format of `Add <project-name> recipe`
- [x] My recipe is reproducible and explains everything needed to run successfully. If my code has external dependencies, I make mention of them.
- [x] My code is easily understandable and/or well-commented.

### CODE REVIEW

**Comment by prefectcboyd** [comment]
Hi Joe,

Any chance of a high level readme or usage / demonstration?
The deploy script itself is well written - we do have some terraform modules that I believe are in process of being re-factored, but this can be useful in the absence of those.

**Comment by chicago-joe** [comment]
> Hi Joe,
> 
> Any chance of a high level readme or usage / demonstration? The deploy script itself is well written - we do have some terraform modules that I believe are in process of being re-factored, but this can be useful in the absence of those.

Ah yes, I had thought about this but wasn't sure if it would be okay for me to modify the readme that @gabcoyne had written. 

I will try to get this out later today; readme should be fairly short and will explain the variable args that a user will need to modify in the deploy script. Cheers!

File: devops/infrastructure-as-code/aws/cli-prefect2-ecs-agent/deploy_ecs.sh [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                "ecs:RegisterTaskDefinition",
+                "ecs:RunTask",
+                "ecs:StopTask",
+                "iam:PassRole",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "logs:DescribeLogGroups",
+                "logs:GetLogEvents",
+                "codecommit:PutFile",
+                "codecommit:GetObjectIdentifier",
```
Comment (gabcoyne) [inline devops/infrastructure-as-code/aws/cli-prefect2-ecs-agent/deploy_ecs.sh:93]: Everything in here except for codecommit I think fits, codecommit access is not required by the ECS agent. Can we remove the codecommit permissions?

**Comment by gabcoyne** [comment]
This recipe is for the CLI, so I'd like to keep the readme based, procedural feel. I like the additional explicit policy additions but I'd like to split them out from the shell script. @prefectcboyd I'd like your thoughts here.

**Review by gabcoyne — CHANGES_REQUESTED** [review]

File: devops/infrastructure-as-code/aws/cli-prefect2-ecs-agent/deploy_ecs.sh (hunk 1 above)
Comment (prefectcboyd) [reply devops/infrastructure-as-code/aws/cli-prefect2-ecs-agent/deploy_ecs.sh:93]: > This recipe is for the CLI, so I'd like to keep the readme based, procedural feel. I like the additional explicit policy additions but I'd like to split them out from the shell script. @prefectcboyd I'd like your thoughts here.

Agreed , and I think you both have valid points -  keeping the original Readme as a CLI based procedural doc.
Then moving this work as a sub-folder to script the process, with a separate readme, to describe usage of this.
Does that make sense?

The net effect would be :

##Current 
cli-prefect2-ecs-agent/
- Readme.md
- prefect-agent-td.json
- trust-policy.json

##Suggested
cli-prefect2-ecs-agent/
- Readme.md
- prefect-agent-td.json
- trust-policy.json
- scripted_deploy/
   - Readme.md
   - deploy_ecs.sh
   - ecs_task_role.json

Comment (serinamarie) [reply devops/infrastructure-as-code/aws/cli-prefect2-ecs-agent/deploy_ecs.sh:93]: @chicago-joe When you get a chance 🙂  ^

**Comment by serinamarie** [comment]
Closing as there has been no activity for over a month. @chicago-joe Feel free to reopen if you wish.

**Pull request closed without merging**